samples/template-operator/operator/controllers/sample_controller.go

```diff
@@ -62,13 +61,18 @@ func (r *SampleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// initReconciler injects the required configuration into the declarative reconciler.
 func (r *SampleReconciler) initReconciler(mgr ctrl.Manager) error {
+	manifestResolver := &ManifestResolver{}
 	return r.Inject(mgr, &v1alpha1.Sample{},
+		declarative.WithManifestResolver(manifestResolver),
 		declarative.WithResourceLabels(map[string]string{"sampleKey": "sampleValue"}),
 		declarative.WithObjectTransform(transform),
```

Comment:
```suggestion
		declarative.WithPostRenderTransform(transform),
```
since this is after the helm rendering, I think we could consider renaming this